app/controllers/histories_controller.rb

```diff
@@ -46,16 +46,24 @@ def check_role
   end
 
   def check_patient
-    @patient = current_user.viewable_patients.find_by(id: params.require(:patient_id))
-    return head :forbidden if @patient.nil?
+    patient_id = params.require(:patient_id)&.to_i
+
+    # Check if Patient ID is valid
+    unless Patient.exists?(patient_id)
+      error_message = "History comment cannot be modified for unknown monitoree with ID: #{patient_id}"
+      render(json: { error: error_message }, status: :bad_request) && return
+    end
+
+    # Check if user has access to patient
+    @patient = current_user.viewable_patients.find_by(id: patient_id)
```

Comment:
```suggestion
    @patient = current_user.viewable_patients.find(patient_id)
```

Reply:
causes this test to fail: https://github.com/SaraAlert/SaraAlert/blob/controller-test-updates/test/controllers/histories_controller_test.rb#L54

Reply:
Oh duh sorry, because `find` throws an error if there are not matches. You want `find_by_id`.

Reply:
updated here: https://github.com/SaraAlert/SaraAlert/pull/900/commits/a3cd41ef431c8db20bb05e0b5d49fac393d0e383